llvm/lib/CodeGen/AsmPrinter/CodeViewDebug.cpp

```diff
@@ -628,6 +628,11 @@ void CodeViewDebug::beginModule(Module *M) {
     // When emitting only compiler information, we may have only NoDebug CUs,
     // which would be skipped by debug_compile_units_begin.
     NamedMDNode *CUs = MMI->getModule()->getNamedMetadata("llvm.dbg.cu");
+    if (CUs->operands().empty()) {
+      Asm = nullptr;
+      return;
+    }
+
```

Comment:
Sorry to code golf this, but I think this would be cleaner if we took the CU variable below and stored it in CodeViewDebug (call it `TheCU` or something) and reused it in the `endModule` invocation. It goes right next to the `CompilerInfoAsm` state. 

Part of what's going on here is that we never came up with a proper strategy for handling full LTO, which is what usually creates a multi-CU situation. The emergent behavior is probably fine, and maybe the best we could do, which is just using the global module information from the first compilation unit in the LTO unit. People probably already rely on that.